Library/Homebrew/cmd/test-bot.rb

```diff
@@ -520,7 +521,7 @@ def formula formula_name
               test "brew", "uninstall", "--force", *unchanged_build_dependencies
               unchanged_dependencies -= unchanged_build_dependencies
             end
-            test "brew", "install", bottle_filename
+            test "brew", "install", "--force-bottle", bottle_filename
```

Comment:
I've got to dash out briefly, but will try and look at it when I come back. Feel free to throw around ideas though, nothing immediately comes to mind that doesn't either alter test-bot behaviour or alter the intended behaviour of the new modified formula check.